lib/Site.js

```diff
@@ -132,16 +133,25 @@ function rejectHandler(reject, error, removeFolders) {
     });
 }
 
-function getGlyphiconsMap() {
+function getIconsMap() {
+  const fontAwesomePath = path.join(__dirname, FONT_AWESOME_PATH);
   const glyphiconsPath = path.join(__dirname, GLYPHICONS_PATH);
+  const fontAwesomeClasses = fs.readFileSync(fontAwesomePath, 'utf8').trimRight().split(/\r?\n/);
   const glyphiconsClasses = fs.readFileSync(glyphiconsPath, 'utf8').trimRight().split(/\r?\n/);
-  return glyphiconsClasses.reduce((glyphiconsMap, glyphiconClass) => {
-    const name = glyphiconClass.replace(/-/g, '_');
-    const html = `<span class='glyphicon ${glyphiconClass}' aria-hidden='true'></span>`;
-    // eslint-disable-next-line no-param-reassign
-    glyphiconsMap[name] = html;
-    return glyphiconsMap;
-  }, {});
+  const iconsMap = {};
+  // Collate Font Awesome Icons
+  fontAwesomeClasses.forEach((fontAwesomeClass) => {
+    const faName = fontAwesomeClass.replace(' fa', '').replace(/-/g, '_');
+    const faHtml = `<span class='${fontAwesomeClass}'></span>`;
+    iconsMap[faName] = faHtml;
+  });
+  // Collate Glyphicons
+  glyphiconsClasses.forEach((glyphiconClass) => {
+    const glyphName = glyphiconClass.replace(/-/g, '_');
+    const glyphHtml = `<span class='glyphicon ${glyphiconClass}' aria-hidden='true'></span>`;
+    iconsMap[glyphName] = glyphHtml;
+  });
```

Comment:
Was wondering whether there's any value in refactoring the code since it seems the logic for "Font Awesome" and "Glyphicons" are almost identical.

Reply:
It is difficult to abstract into a single function as they have different algorithms for extracting their variable name and the html formatting is also different.